File: templates/ui-components/ios-objc/TemplateManager.m
//  Created by react-native-create-bridge

#import <Foundation/Foundation.h>
#import "{{template}}.h"
#import "{{template}}Manager.h"

// import RCTBridge
#if __has_include(<React/RCTBridge.h>)
#import <React/RCTBridge.h>
#elif __has_include(“RCTBridge.h”)
#import “RCTBridge.h”
#else
#import “React/RCTBridge.h” // Required when used as a Pod in a Swift project
#endif

@implementation {{template}}Manager

@synthesize bridge = _bridge;

// Export a native module
// https://facebook.github.io/react-native/docs/native-modules-ios.html
RCT_EXPORT_MODULE();

// Return the native view that represents your React component
- (UIView *)view
{
  return [[{{template}} alloc] initWithEventDispatcher:self.bridge.eventDispatcher];
}

RCT_EXPORT_VIEW_PROPERTY(exampleProp, NSString)

// Export constants
// https://facebook.github.io/react-native/releases/next/docs/native-modules-ios.html#exporting-constants
- (NSDictionary *)constantsToExport
{
  return @{
           @"EXAMPLE": @"example"
         };
}

@end

File: templates/ui-components/ios-swift/TemplateManager.swift
//  Created by react-native-create-bridge

import Foundation
import UIKit

@objc({{template}})
class {{template}}Manager : RCTViewManager {
  // Return the native view that represents your React component
  override func view() -> UIView! {
    return {{template}}View()
  }

  // Export constants to use in your native module
  override func constantsToExport() -> [String : Any]! {
    return ["EXAMPLE_CONSTANT": "example"]
  }

}
